projects/project-2/readings/Project 2 - SPARQL library

```diff
@@ -0,0 +1,297 @@
+Assignment: SPARQL Queries for Ontology Evaluation
+
+This assignment outlines a series of SPARQL queries designed for evaluating and analyzing ontology structure and quality. 
+Each query addresses specific aspects of ontology evaluation, identifying potential issues or improvements necessary to enhance clarity and consistency.
+
+Detect Properties Without Domain or Range (level 8 - 0 points)
+Identifying Disjoint Classes with Common Instances (level 7 - 2 points)
+Detecting Unused Object Properties (level 7 - 2 points)
+Finding Classes with Conflicting Disjointness and Subclass Relationships (level 6 - 3 points)
+Identifying Classes with Different Labeling Standards (level 6 - 3 points)
+Identifying Potentially Overspecified Class Definitions (level 5 - 5 points)
+Detecting Equivalent Properties with Different Ranges  (level 5 - 5 points)
+Analyzing Class Hierarchies for Inconsistent Type Definitions (level 5 - 5 points)
+Detecting Potentially Misplaced Inverse Object Properties (level 5 - 5 points)
+Identifying Classes with No Properties (Level 4 - 10 points)
+Identifying Classes Without Labels (Level 4 - 10 points)
+Query to Detect Missing Annotations (Labels or Comments) Level 4 - 10 points
+Detecting Properties Shared Across Multiple CCO Classes Level 3 - 20 points
+Detecting Inconsistent Property Ranges Across Subclass Hierarchies Level 3 - 20 points
+
+1. Detect Classes Without Any Instances (Level 8 - 0 points)
+
+Title: Empty Class Detection
+Constraint Description: Identifies classes that have no instances associated with them.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class
+WHERE {
+  ?class a rdfs:Class .
+  FILTER NOT EXISTS {
+    ?instance a ?class .
+  }
+}
+
+Purpose: ​This query retrieves all classes that do not have any instances associated with them.​ 
+It uses FILTER NOT EXISTS to ensure that only classes without any defined instances are selected,helping to identify potentially unused or unnecessary classes in the ontology.
+
```

Comment:
One way to improve this query is to include classes that are implicitly defined through subclassing. This way, classes defined by rdfs:subClassOf relationships but not explicitly typed as rdfs:Class would also be considered. you could add an alternative condition, like:

{
  ?class a rdfs:Class
}
UNION
{
  ?class rdfs:subClassOf ?superclass
}

Reply:
Okay, thank you Jisoo :)